echo-telemetry/src/main/java/com/netflix/spinnaker/echo/telemetry/TelemetryEventListener.java

```diff
@@ -70,26 +48,27 @@
   private static final JsonFormat.Printer JSON_PRINTER =
       JsonFormat.printer().includingDefaultValueFields();
 
-  private static final ObjectMapper objectMapper = EchoObjectMapper.getInstance();
-
   private final TelemetryService telemetryService;
 
-  private final TelemetryConfig.TelemetryConfigProps telemetryConfigProps;
-
   private final CircuitBreakerRegistry registry;
 
+  private final List<TelemetryEventDataProvider> dataProviders;
+
   @Autowired
   public TelemetryEventListener(
       TelemetryService telemetryService,
-      TelemetryConfig.TelemetryConfigProps telemetryConfigProps,
-      CircuitBreakerRegistry registry) {
+      CircuitBreakerRegistry registry,
+      List<TelemetryEventDataProvider> dataProviders) {
     this.telemetryService = telemetryService;
-    this.telemetryConfigProps = telemetryConfigProps;
     this.registry = registry;
+    this.dataProviders = dataProviders;
   }
 
   @Override
   public void processEvent(Event event) {
+
+    // These preconditions are also guaranteed to TelemetryEventDataProvider, so don't change them
+    // without checking the implementations.
```

Comment:
Not changed here, but while here maybe change `log.debug("Telemetry not sent: type '{}' not whitelisted ", eventType);` a few lines down to "type not configured" or something similar?

Reply:
Great catch, thanks!